src/WebTestAssertionsTrait.php

```diff
@@ -63,32 +95,152 @@ public static function assertPageTitleContains(string $expectedTitle, string $me
+    {
+        /** @var PantherClient $client */
+        $client = self::getClient();
+        $crawler = $client->waitForStaleness($locator);
+        self::assertSelectorNotExists($locator);
+
+        return $crawler;
+    }
+
     public static function assertSelectorIsVisible(string $locator): void
     {
         $element = self::findElement($locator);
         self::assertTrue($element->isDisplayed(), 'Failed asserting that element is visible.');
     }
 
+    public static function assertSelectorWillBeVisible(string $locator): Crawler
+    {
+        /** @var PantherClient $client */
+        $client = self::getClient();
+        $crawler = $client->waitForVisibility($locator);
+        self::assertSelectorIsVisible($locator);
+
+        return $crawler;
+    }
+
     public static function assertSelectorIsNotVisible(string $locator): void
     {
         $element = self::findElement($locator);
         self::assertFalse($element->isDisplayed(), 'Failed asserting that element is not visible.');
     }
 
+    public static function assertSelectorWillNotBeVisible(string $locator): Crawler
+    {
+        /** @var PantherClient $client */
+        $client = self::getClient();
+        $crawler = $client->waitForInvisibility($locator);
+        self::assertSelectorIsNotVisible($locator);
+
+        return $crawler;
+    }
+
+    public static function assertSelectorWillContain(string $locator, string $text): Crawler
+    {
+        /** @var PantherClient $client */
+        $client = self::getClient();
+        $crawler = $client->waitForElementToContain($locator, $text);
+        self::assertSelectorTextContains($locator, $text);
+
+        return $crawler;
+    }
+
+    public static function assertSelectorWillNotContain(string $locator, string $text): Crawler
+    {
+        /** @var PantherClient $client */
+        $client = self::getClient();
+        $crawler = $client->waitForElementToNotContain($locator, $text);
+        self::assertSelectorTextNotContains($locator, $text);
+
+        return $crawler;
+    }
+
     public static function assertSelectorIsEnabled(string $locator): void
     {
         $element = self::findElement($locator);
         self::assertTrue($element->isEnabled(), 'Failed asserting that element is enabled.');
     }
 
+    public static function assertSelectorWillBeEnabled(string $locator): Crawler
+    {
+        /** @var PantherClient $client */
+        $client = self::getClient();
+        $crawler = $client->waitForEnabled($locator);
+        self::assertSelectorAttributeToContain($locator, 'disabled');
+
+        return $crawler;
+    }
+
     public static function assertSelectorIsDisabled(string $locator): void
     {
         $element = self::findElement($locator);
         self::assertFalse($element->isEnabled(), 'Failed asserting that element is disabled.');
     }
 
+    public static function assertSelectorWillBeDisabled(string $locator): Crawler
```

Comment:
I wonder if it's a good API or not to return the crawler here. Usually, assertions have no return value. Shouldn't users call `Client::getCrawler()` if they want to access the crawler?

Reply:
crawler is updated so I remove them everywhere. Should we keep them as return of `waitForX` methods?

Reply:
Yes it's ok for `waitFor` method imho (they aren't assertions).